x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -286,17 +287,25 @@ export type PreConfiguredActionConnector = Omit<
   'config' | 'secrets'
 > & {
   isPreconfigured: true;
+  isSystemAction: false;
 };
 
 export type UserConfiguredActionConnector<Config, Secrets> = ActionConnectorProps<
   Config,
   Secrets
 > & {
   isPreconfigured: false;
+  isSystemAction: false;
+};
+
+export type SystemAction = Omit<ActionConnectorProps<never, never>, 'config' | 'secrets'> & {
+  isSystemAction: true;
+  isPreconfigured: false;
```

Comment:
If you imagine we end up changing `isPreconfigured` to `isInMemory` or similiar, we'd end up probably wanting `isPreconfigured` to be `true`, whenever `isSystemAction` is true.  That way, we won't have to change any of our current `isPreconfigured` checks, that are typically to check if it's persisted as an SO or not.

Reply:
I don't know how this will play out in the UI yet. I suspect that we would like to have the separation. I will keep your suggestion in mind when the time comes. In the backend, we should have the separation as we treat the preconfigured and the system actions a bit differently.